scripts/dev/ensure_unique_html_tables.py

```diff
@@ -0,0 +1,125 @@
+#     obtained under this License and may not use a different name for the
+#     software. Except as specifically required in this Section (4), Licensee
+#     shall not use in a company name, a product name, in advertising,
+#     publicity, or other promotional activities any name, trade name,
+#     trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or
+#     confusingly similar designation, without the U.S. Department of Energy's
+#     prior written consent.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import re
+from collections import Counter
+from pathlib import Path
+
+RE_FULLNAME = re.compile(r"<!-- FullName:(?P<activeReportName>[^_]+)_(?P<activeForName>[^_]+)_(?P<subtitle>[^_]+)-->")
+
+
+def format_row(row, col_widths):
+    return "| " + " | ".join(f"{str(cell):<{col_widths[i]}}" for i, cell in enumerate(row)) + " |"
+
+
+def ensure_unique_html_tables(html_path: Path):
+    """
+    Ensure that each HTML table in the report has a unique name.
+    If duplicates are found, append a number to the duplicate names.
+    """
+    content = html_path.read_text()
+
+    matches = RE_FULLNAME.findall(content)
+    if not matches:
+        raise ValueError("No HTML tables found with FullName pattern")
+
+    # Filter only duplicates (count > 1)
+    duplicates = [
+        (activeReportName, activeForName, subtitle, count)
+        for (activeReportName, activeForName, subtitle), count in Counter(matches).items()
+        if count > 1
+    ]
+
+    # Include header row in width calculation
+    if duplicates:
+        print(f"Found {len(duplicates)} duplicate HTML table(s) based on FullName\n")
+        header = [("activeReportName", "activeForName", "subtitle", "count")]
+        rows = header + duplicates
+        # Compute max width for each column
+        col_widths = [max(len(str(row[i])) for row in rows) for i in range(len(header[0]))]
+
+        print(format_row(rows[0], col_widths))
+        print("|-" + "-|-".join("-" * w for w in col_widths) + "-|")
+        for row in duplicates:
+            print(format_row(row, col_widths))
+
+        exit(1)
+    else:
+        print("No duplicates found.")
+        exit(0)
```

Comment:
Pretty simple: parse the FUllName stuff into 3-tuples. Counter to find duplicates, format as markdown table, then exit 1.

```
$ python EnergyPlus/scripts/dev/ensure_unique_html_tables.py EnergyPlus-build-release/testfiles/SingleFamilyHouse_HP_Slab_Dehumidification/
Found 1 duplicate HTML table(s) based on FullName

| activeReportName       | activeForName   | subtitle     | count |
|------------------------|-----------------|--------------|-------|
| Initialization Summary | Entire Facility | Material:Air | 2     |
(py312)(3.2.2)julien@EnergyPlus-build-release$ echo $?
1

```